drivers/flash/flash_stm32_qspi.c

```diff
@@ -717,8 +717,22 @@ static int setup_pages_layout(const struct device *dev)
 }
 #endif /* CONFIG_FLASH_PAGE_LAYOUT */
 
-static int qspi_program_addr_4b(const struct device *dev)
+static int qspi_program_addr_4b(const struct device *dev, bool write_enable)
 {
+	int ret;
+
+	/* Send write enable command, if required */
+	if (write_enable) {
+		QSPI_CommandTypeDef cmd_write_en = {
+			.Instruction = SPI_NOR_CMD_WREN,
+			.InstructionMode = QSPI_INSTRUCTION_1_LINE,
+		};
+		ret = qspi_send_cmd(dev, &cmd_write_en);
+		if (ret != 0) {
+			return ret;
+		}
```

Comment:
Why not using `qspi_write_enable()` ?

Reply:
qspi_write_enable also polls the status register. In flash_stm32_qspi_write for example, there is also a dedicated extra command send for it, I guess to avoid waiting for the status, which is done later. I guess in the end it doesn't matter, but if we take the specification literally "issue write enable instruction 06h, then issue instruction B7h", then we shouldn't poll the status register after sending the write enable instruction, so the current code should be right. But I can replace it with the qspi_write_enable as well, if you think this is better. For the code it would look nicer.

Reply:
Maybe a "status" parameter for the qspi_write_enable function would be useful, then the caller of the function can decide if the status should be tested after the command. This could simplify some of the other code in flash_stm32_qspi_write as well. But not sure if this really makes it much more readable, and the C compiler probably optimizes it all out anyway.
What is the policy for PRs in Zephyr? Is it ok to refactor and add features at the same time?

Reply:
> What is the policy for PRs in Zephyr? Is it ok to refactor and add features at the same time?

As long as you split refactoring and additions in different commits it is ok (of course a complete driver overhaul may deserve a specific PR, but this is not the case here).

> the specification literally "issue write enable instruction 06h, then issue instruction B7h", then we shouldn't poll the status register after sending the write enable instruction, so the current code should be right.

Ok, I guess we can keep it that way then.